### PR TITLE
Fixup python package metadata for PyPI publishing.

### PR DESCRIPTION
**Pull request by ScottTodd**
Note: I haven't tested this. Not sure if there's a better way than #testinprod though (maybe testpypi, but that doesn't help with this PR).

---

See https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/#classifiers and https://pypi.org/classifiers/ - the "license" classifier must match one in that list, so I replaced "Apache License" with "Apache Software License".

I also alphabetized the classifiers and dropped "Operating System :: OS Independent" from the packages that are not actually OS independent (as they include native C/C++ code that is not platform independent and we may never publish Windows wheels for those packages anyways - see https://github.com/google/iree/issues/6417).

### CODE REVIEW

File: bindings/python/iree/runtime/setup.py.in [hunk 1]
```diff
@@ -22,8 +22,10 @@ setup(
     description="IREE Python Runtime Components",
     long_description=README,
     long_description_content_type="text/markdown",
-    license="Apache-2.0",
+    license="Apache 2.0",
```
Comment (ScottTodd) [inline bindings/python/iree/runtime/setup.py.in:25]: These changes don't matter. JAX uses `Apache-2.0` but TensorFlow uses `Apache 2.0`

File: bindings/python/iree/runtime/setup.py.in [hunk 2]
```diff
@@ -22,8 +22,10 @@ setup(
...
     classifiers=[
+        "Development Status :: 3 - Alpha",
+        "License :: OSI Approved :: Apache Software License",
```
Comment (ScottTodd) [inline bindings/python/iree/runtime/setup.py.in:28]: For reference:

https://github.com/google/jax/blob/main/setup.py
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/pip_package/setup.py

File: bindings/python/iree/runtime/setup.py.in (hunk 1 above)
Comment (ScottTodd) [reply bindings/python/iree/runtime/setup.py.in]: (Reverted those)

**Review by stellaraccident — APPROVED** [review]
Thanks! There's not a good way to test this (that is worth doing)

**Pull request merged**